blocks/ai-chat/src/complete-chat.tsx

```diff
@@ -0,0 +1,486 @@
+import { EntityId } from "@blockprotocol/graph";
+import { useGraphBlockModule } from "@blockprotocol/graph/react";
+import { useServiceBlockModule } from "@blockprotocol/service/react";
+import { Box, Collapse } from "@mui/material";
+import {
+  FunctionComponent,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { TransitionGroup } from "react-transition-group";
+
+import { ChatMessage } from "./complete-chat/chat-message";
+import {
+  ChatModelId,
+  defaultChatModelId,
+  isChatModelId,
+} from "./complete-chat/chat-model-selector";
+import { ChatTextField } from "./complete-chat/chat-textfield";
+import { ExamplePrompts } from "./complete-chat/example-prompts";
+import { Header } from "./complete-chat/header";
+import {
+  defaultSystemPromptId,
+  isSystemPromptId,
+  SystemPromptId,
+  systemPrompts,
+} from "./complete-chat/system-prompt-selector";
+import {
+  IncompleteOpenAiAssistantMessage,
+  OpenAIChatMessage,
+} from "./complete-chat/types";
+import { AIChatBlock } from "./types/generated/ai-chat-block";
+import {
+  createRequestMessageEntityMethod,
+  createResponseMessageEntityMethod,
+  propertyTypeBaseUrls,
+} from "./types/graph";
+
+let requestCounter = 0;
+
+export type RequestId = `req_${number}`;
+
+export const createRequestId = (): RequestId => `req_${requestCounter++}`;
+
+let responseCounter = 0;
+
+export type ResponseId = `res_${number}`;
+
+export const createResponseId = (): ResponseId => `res_${responseCounter++}`;
```

Comment:
Suggestion: you can use `uuid` instead of the counter,

Reply:
I could, but this is an added dependency so decided to go without it. Any benefit for adding it now that you can think of?

Reply:
Just code simplification, really. Happy to keep it as is

Reply:
Fair enough, have addressed in https://github.com/hashintel/hash/pull/2299/commits/43f7e4ef5f39cde0c158b43f43bae256776cb2df